[ADD] general: integrations google translate (#8202)
odoo/documentation: content/applications/general/integrations/translate.rst

================
Google Translate
================

*Google Translate* can be used to translate user generated text in the Odoo chatter. To translate a
user's text from another language click the :guilabel:`three dot` menu (ellipsis) to the right of
the chatter. Then select :guilabel:`Translate`. The content translates to the *language* set on the
user's preferences.

.. seealso::
   :ref:`language/change-user-language`

.. image:: translate/google-translate.png
   :align: center
   :alt: Google Translate present in an Odoo database's chatter.

Integration is simple with some configuration on *Google's Developer Console* and copying and
pasting an *API key* into Odoo.

Google API Console
==================

A majority of the setup for integrating *Google Translate* into Odoo is done with *Google*. Once the
following processes are complete an :guilabel:`API key` will be created to input in Odoo.

.. seealso::
   `Google Translate setup on Google <https://cloud.google.com/translate/docs/setup>`_

Create a new project
--------------------

> To get started, go to the `Google API Console <https://console.developers.google.com>`_. Log in with
> the *Google Workspace* account if there is one, otherwise log in with a personal Gmail account (this
> should match the email address that has billing attached to it).

Reviewer: ```suggestion
To get started, go to the `Google API Console <https://console.developers.google.com>`_. Then, log
in with a *Google Workspace* account, if there is one. If not, log in with a personal Gmail account
(this should match the email address that has billing attached to it).
```

> Next, click :guilabel:`Create Project` on the far right of the :guilabel:`OAuth consent screen`.

Reviewer: ```suggestion
Next, click :guilabel:`Create Project` on the far-right of the :guilabel:`OAuth consent screen`.
```

.. tip::
   If the *Google Developer Console* has existing projects, click the drop-down menu next to the

>    :guilabel:`Google Cloud` icon and a pop-over appears. Next, click :guilabel:`New Project` in the

Reviewer: ```suggestion
   :guilabel:`Google Cloud` icon, and a pop-over window appears. Next, click :guilabel:`New Project` in the
```

>    top right of the pop-over.

Reviewer: ```suggestion
   top-right of the pop-over window.
```

> On the :menuselection:`New Project` screen, rename the :guilabel:`Project name` to `Odoo Translate`

Reviewer: ```suggestion
On the :menuselection:`New Project` screen, rename the :guilabel:`Project name` to `Odoo Translate`,
```

================
Google Translate
================

*Google Translate* can be used to translate user generated text in the Odoo chatter. To translate a

> user's text from another language click the *three dot* menu to the right of the chatter. Then

Reviewer 2: I think *three dot* needs a guilabel, also I would confirm the actual name in the inspect element menu. (I believe it is *ellipsis* but I am not sure, let me know and I will add it to the style guide doc)

Author: You're right it is an ellipsis-- but I fear our customers won't understand this phrase. I'm going to put in parentheses.

select :guilabel:`Translate`. The language will translate to the *language* set on the user's
preferences.

.. seealso::
   :ref:`language/change-user-language`

.. image:: translate/google-translate.png
   :align: center
   :alt: Google Translate present in an Odoo database's chatter.

> Integration is simple with some configuration on *Google's Developer Console* and copy and pasting

Reviewer 2: ```suggestion
Integration is simple with some configuration on *Google's Developer Console* and copying and pasting
```

Create a new project
--------------------

To get started, go to the `Google API Console <https://console.developers.google.com>`_. Log in with
the *Google Workspace* account if there is one, otherwise log in with a personal Gmail account (this
should match the email address that has billing attached to it).

After that, click on :guilabel:`Create Project`, located on the far right of the :guilabel:`OAuth
consent screen`.

.. tip::
   If the *Google Developer Console* has existing projects, click the drop-down menu next to the
   :guilabel:`Google Cloud` icon and a popover will appear. Next, click :guilabel:`New Project` in
   the top right of the popover.

On the :menuselection:`New Project` screen, rename the :guilabel:`Project name` to `Odoo Translate`
and browse for the :guilabel:`Location`. Set the :guilabel:`Location` as the *Google Workspace
organization*. If a personal Gmail account is being used, then leave the :guilabel:`Location` as
:guilabel:`No Organization`.

.. image:: translate/new-project.png
   :align: center
   :alt: Project Name and Location for Google OAuth.

Click on :guilabel:`Create` to finish this step.

API Library
-----------

> Next the :guilabel:`Cloud Translation API` needs to be installed on this newly created project.

Reviewer 2: ```suggestion
Next, the :guilabel:`Cloud Translation API` needs to be installed on this newly created project.
```

First click :guilabel:`Library` in the left menu. The search the term `Cloud Translation API` and
click into the result. This should be a *Google Enterprise API* labeled `Cloud Translation API`.
Click :guilabel:`Enable` to install the library on this project.

.. important::
   Using the *Google Translate* API requires a current billing account with `Google
   <https://https://myaccount.google.com/>`_.

Once a billing account is setup with *Google* and the library is enabled, click :guilabel:`Manage`
to finish configuration on the API.

Create Credentials
------------------

> Now that the project is set up and library is enabled, credentials should be created. These include

Reviewer 2: ```suggestion
Now that the project is set up and the library is enabled, credentials need to be created. These include
```
Changing this to "need" unless it is an optional step, in which case I would add a note admonition block to make that clear

================
Google Translate
================

> *Google Translate* can be used to translate user generated text in the Odoo chatter. To translate a
> user's text from another language, click the :guilabel:`...(three dot)` icon menu to the right of
> the chatter. Then, select :guilabel:`Translate`. The content translates to the *language* set on the
> user's preferences.
> 
> .. seealso::
>    :ref:`language/change-user-language`
> 
> .. image:: translate/google-translate.png
>    :align: center
>    :alt: Google Translate present in an Odoo database's chatter.
> 

Reviewer 3: Suggesting to move this content after the **Odoo set up** section, so that the document flow follows the traditional _overview --> configuration --> using_ flow.
```suggestion
*Google Translate* can be used to translate user generated text in the Odoo chatter.
```

> Integration is simple with some configuration on *Google's Developer Console*, along with copying
> and pasting an *API key* into Odoo.

Reviewer 3: This sentence is essentially saying the same thing as the opening paragraph in the **Google API console** section, I think you can just lead with that, but it is up to you!

```suggestion
```

Google API console
==================

> A majority of the setup for integrating *Google Translate* into Odoo is done with *Google*. Once the
> following processes are complete an :guilabel:`API key` is created to input in Odoo.

Reviewer 3: Slight rewording for specificity and no need for a guilabel
```suggestion
A majority of the setup for integrating *Google Translate* into Odoo is done with the *Google API
console*. Once the following processes are complete, an *API key* is created to input in Odoo.
```

.. seealso::
   `Google Translate setup on Google <https://cloud.google.com/translate/docs/setup>`_

Create a new project
--------------------

To get started, go to the `Google API Console <https://console.developers.google.com>`_. Then, log
in with a *Google Workspace* account, if there is one. If not, log in with a personal Gmail account
(this should match the email address that has billing attached to it).

Next, click :guilabel:`Create Project` on the far-right of the :guilabel:`OAuth consent screen`.

.. tip::
   If the *Google Developer Console* has existing projects, click the drop-down menu next to the
   :guilabel:`Google Cloud` icon, and a pop-over window appears. Next, click :guilabel:`New Project`
   top-right of the pop-over window.

> On the :menuselection:`New Project` screen, rename the :guilabel:`Project name` to `Odoo Translate`,
> and browse for the :guilabel:`Location`. Set the :guilabel:`Location` as the *Google Workspace

Reviewer 3: ```suggestion
On the :guilabel:`New Project` screen, rename the :guilabel:`Project name` to `Odoo Translate`, and
browse for the :guilabel:`Location`. Set the :guilabel:`Location` as the *Google Workspace
```

organization*. If a personal Gmail account is being used, leave the :guilabel:`Location` as
:guilabel:`No Organization`.

.. image:: translate/new-project.png
   :align: center
   :alt: Project Name and Location for Google OAuth.

Click on :guilabel:`Create` to finish this step.

> API Library

Reviewer 3: ```suggestion
API library
```

-----------

> Next, the :guilabel:`Cloud Translation API` needs to be installed on this newly-created project.
> To do that, click :guilabel:`Library` in the left menu. Then, search the term `Cloud Translation
> API`, and click into the result. This should be a *Google Enterprise API* labeled `Cloud Translation
> API`.

Reviewer 3: Proper tag use
```suggestion
Next, the *Cloud Translation API* needs to be installed on this newly-created project. To do that,
click :menuselection:`Library` in the left menu. Then, search the term `Cloud Translation API`, and
click into the result. This should be a *Google Enterprise API* labeled :guilabel:`Cloud Translation
API`.
```

Click :guilabel:`Enable` to install the library on this project.

.. important::
   Using the *Google Translate* API **requires** a current billing account with `Google
   <https://https://myaccount.google.com/>`_.

Once a billing account is setup with *Google* and the library is enabled, click :guilabel:`Manage`
to finish configuration on the API.

> Create Credentials

Reviewer 3: Sentence case headings please
```suggestion
Create credentials
```

------------------

> Now that the project is set up, and the library is enabled, credentials **must** be created. This
> includes the *API key*.

Reviewer 3: Suggestion to be specific here, as long as this makes sense
```suggestion
Now that the project is set up, and the *Cloud Translation API* is enabled, credentials **must** be
created. This includes the *API key*.
```

> To begin this process, click :guilabel:`Credentials` in the left sidebar menu.

Reviewer 3: I believe a menuselection is a better fit here, since it is a menu the user is clicking on
```suggestion
To begin this process, click :menuselection:`Credentials` in the left sidebar menu.
```

Then, click :guilabel:`Create Credentials` in the top menu, and select :guilabel:`API key` from the
drop-down menu.

.. image:: translate/api-key.png
   :align: center
   :alt: Create an API key in the Google developer console.

Copy the :guilabel:`API key` for use in the next section.

> .. warning::

Reviewer 3: I feel an important admonition is more fitting here
```suggestion
.. important::
```

   For security purposes, the usage of the *API key* can be restricted.

>    To do that, go to the :guilabel:`API restrictions` by clicking on :guilabel:`Edit API key` in the
>    pop-over window, or on the listed :guilabel:`API key` on the :guilabel:`Credentials` page. From
>    here, key restrictions can be set. This includes setting an application to restrict the use of
>    the API key, and whether this *API key* can call any API.

Reviewer 3: Tag use and specificity 
```suggestion
   To do that, go to the *API restrictions* by clicking on :guilabel:`Edit API key` in the pop-over
   window, or by clicking on the listed API key on the :guilabel:`Credentials` page. From here, key
   restrictions can be set. This includes setting an application to restrict the use of the API key,
   and whether this API key can call any API.
```

   It is recommended that the Odoo *Translate API* be restricted to **only** allow requests from the

>    configured database and to the `Cloud Translation API`.

Reviewer 3: Specificity and formatting
```suggestion
   configured Odoo database and to the *Cloud Translation API*.
```

>    To add the website restriction, click :guilabel:`Websites`, under the :guilabel:`Set an
>    application restriction`. Then, enter the address of the database *Google Translate* is being
>    used in, by clicking on :guilabel:`Add`. Lastly, add the :abbr:`URL (Uniform Resource Locator)`,
>    and click :guilabel:`Done`.

Reviewer 3: Are :guilabel:s missing here for Address and URL fields?
```suggestion
   To add the website restriction, click :guilabel:`Websites`, under :guilabel:`Set an application
   restriction`. Then, enter the :guilabel:`Address` of the database *Google Translate* is being
   used in, by clicking on :guilabel:`Add`. Lastly, add the :guilabel:`URL`, and click
   :guilabel:`Done`.
```

Author: No they are not missing

   To restrict use of the key to a selected API, first, select :guilabel:`Restrict key`, under the
   :guilabel:`API restrictions` section. Then use the drop-down menu to choose the API being

>    configured (`Cloud Translation API`).

Reviewer 3: formatting – literal formatting is reserved for user inputs
```suggestion
   configured (*Cloud Translation API*).
```

.. tip::

>    - Save the API Key: copy the API key and securely store it.
>    - Do **not** share it publicly or expose it in client-side code.

Reviewer 3: Capitalization and specificity 
```suggestion
   - Save the API key: copy the API key and store it somewhere secure.
   - Do **not** share the API key publicly or expose it in client-side code.
```

> Odoo set up
> ===========

Reviewer 3: Suggestion to phrase this section
```suggestion
Odoo configuration
==================
```

> To access the integration in Odoo, navigate to the :menuselection:`Settings app --> Discuss section
> --> Message Translation`. Enter the :guilabel:`API key` into the field labeled :guilabel:`Paste your
> API key`. Then, :guilabel:`Save` the settings, and *Google Translate* can be used in any chatter
> throughout the database.

Reviewer 3: The field name is **Message Translation** and the palceholder text within the field is "Paste your API key".
```suggestion
To access the integration in Odoo, navigate to the :menuselection:`Settings app --> Discuss
section`. Enter the API key into the field labeled :guilabel:`Message Translation`. Then,
:guilabel:`Save` the settings, and *Google Translate* can be used in any chatter throughout the
database.
```

.. image:: translate/odoo-config.png
   :align: center

>    :alt: Odoo configuration of the API key from the *Google Developer Console*.

Reviewer 3: Suggestion to move the intro copy down to its own section "Translate chatter" now that all is configured
```suggestion
   :alt: Odoo configuration of the API key from the *Google Developer Console*.
   
Translate chatter
=================

To translate a user's text from another language, click the :guilabel:`... (three dot)` icon menu to
the right of the chatter. Then, select :guilabel:`Translate`. The content translates to the
*language* set on the user's preferences.

.. image:: translate/google-translate.png
   :align: center
   :alt: Google Translate present in an Odoo database's chatter.

.. seealso::
   :ref:`language/change-user-language`
   
```
